src/Schemes/AffineSchemes/Morphisms/Methods.jl

```diff
@@ -0,0 +1,148 @@
+projections.
+"""
+function fiber_product(
+    f::SpecMor{SpecType, SpecType, <:Any},
+    g::SpecMor{SpecType, SpecType, <:Any}
+  ) where {SpecType<:StdSpec}
+  Y = domain(f)
+  X = codomain(f)
+  X == codomain(g) || error("maps need to have the same codomain")
+  Z = domain(g)
+  YxZ, pY, pZ = product(Y, Z)
+  RX = ambient_ring(X)
+  W = subscheme(YxZ, [pullback(pY)(pullback(f)(x)) - pullback(pZ)(pullback(g)(x)) for x in gens(RX)])
+  return W, restrict(pY, W, Y, check=false), restrict(pZ, W, Z, check=false)
+end
+
+
+
+###########################################################
+# (2) The direct product of two affine schemes
+###########################################################
+
+@Markdown.doc """
+    product(X::AbsSpec, Y::AbsSpec)
+    
+Returns a triple ``(X×Y, p₁, p₂)`` consisting of the product ``X×Y`` over 
+the common base ring ``𝕜`` and the two projections ``p₁ : X×Y → X`` and
+``p₂ : X×Y → Y``.
+"""
+function product(X::AbsSpec, Y::AbsSpec;
+    change_var_names_to::Vector{String}=["", ""]
+  )
+  base_ring(X) == base_ring(Y) || error("schemes are not defined over the same base ring")
+  Xstd = standard_spec(X)
+  Ystd = standard_spec(Y)
+  XxY, prX, prY = product(Xstd, Ystd, change_var_names_to=change_var_names_to)
+  return XxY, compose(prX, SpecMor(Xstd, X, gens(OO(Xstd)))), compose(prY, SpecMor(Ystd, Y, gens(OO(Ystd))))
+end
+
+function product(X::StdSpec, Y::StdSpec;
+    change_var_names_to::Vector{String}=["", ""]
+  )
+  K = OO(X)
+  L = OO(Y) 
+  V = localized_ring(K)
+  W = localized_ring(L)
+  R = base_ring(K)
+  S = base_ring(L)
+  k = base_ring(R)
+  k == base_ring(S) || error("varieties are not defined over the same field")
+
+  m = length(gens(R))
+  n = length(gens(S))
+  new_symb = Symbol[]
+  if length(change_var_names_to[1]) == 0
+    new_symb = symbols(R)
+  else 
+    new_symb = Symbol.([change_var_names_to[1]*"$i" for i in 1:ngens(R)])
+  end
+  if length(change_var_names_to[2]) == 0
+    new_symb = vcat(new_symb, symbols(S))
+  else 
+    new_symb = vcat(new_symb, Symbol.([change_var_names_to[2]*"$i" for i in 1:ngens(S)]))
+  end
+  RS, z = PolynomialRing(k, new_symb)
+  inc1 = hom(R, RS, gens(RS)[1:m])
+  inc2 = hom(S, RS, gens(RS)[m+1:m+n])
+  IX = ideal(RS, inc1.(gens(modulus(quotient_ring(OO(X))))))
+  IY = ideal(RS, inc2.(gens(modulus(quotient_ring(OO(Y))))))
+  UX = MPolyPowersOfElement(RS, inc1.(denominators(inverted_set(OO(X)))))
+  UY = MPolyPowersOfElement(RS, inc2.(denominators(inverted_set(OO(Y)))))
+  XxY = Spec(RS, IX + IY, UX*UY)
+  pr1 = SpecMor(XxY, X, gens(RS)[1:m], check=false)
+  pr2 = SpecMor(XxY, Y, gens(RS)[m+1:m+n], check=false)
+  return XxY, pr1, pr2
+end
+
+
+
+###########################################################
+# (3) Simplify an affine scheme
+###########################################################
+
+@Markdown.doc """
+    simplify(X::AbsSpec{<:Field})
+
+Given an affine scheme ``X`` with coordinate ring ``R = 𝕜[x₁,…,xₙ]/I`` 
```

Comment:
This changes the ambient scheme and should be documented is this way -- maybe by adding a phrase explaining this on the scheme level.

Reply:
If we are dealing with embedded schemes, then we can be honest about this change, which is very useful and needed. Right now, we hide the embedding... This needs to be discussed.